modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -91,6 +91,23 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
     });
   };
 
+  const toggleCaption = () => {
+    getSelectionStartCellOrCaption(editor).each((cellOrCaption) => {
+      TableLookup.table(cellOrCaption, isRoot).each((table) => {
+        Selectors.one('caption', table).fold(
```

Comment:
I just realised this won't work as expected because it looks all the way down the dom tree. So if you have a tested table it'll find the caption in that potentially. As such this should be using `SelectorFind.child` instead which will only look at the direct children.

- [ ] Fix this so it won't find captions in nested tables
- [ ] Consider adding a test for this case please.